src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -24,37 +23,39 @@ namespace Microsoft.Azure.Templates.Analyzer.Cli
     /// </summary>
     internal class CommandLineParser
     {
-        RootCommand rootCommand;
-
-        private readonly TemplateAnalyzer templateAnalyzer;
-        private readonly SummaryLoggerProvider summaryLoggerProvider;
+        private readonly IReadOnlyList<string> validSchemas = new List<string> {
+            "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+            "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+            "https://schema.management.azure.com/schemas/2018-05-01/subscriptionDeploymentTemplate.json#",
+            "https://schema.management.azure.com/schemas/2019-08-01/tenantDeploymentTemplate.json#",
+            "https://schema.management.azure.com/schemas/2019-08-01/managementGroupDeploymentTemplate.json#"
+        }.AsReadOnly();
+
+        private readonly IReadOnlyList<string> validTemplateProperties = new List<string> {
+            "contentVersion",
+            "apiProfile",
+            "parameters",
+            "variables",
+            "functions",
+            "resources",
+            "outputs",
+        }.AsReadOnly();
+
+        private const string defaultConfigFileName = "configuration.json";
+
+        private RootCommand rootCommand;
+        private TemplateAnalyzer templateAnalyzer;
+
+        private IReportWriter reportWriter;
+        private ILogger logger;
+        private SummaryLogger summaryLogger;
  
-        string[] validSchemas = {
-               "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
-               "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
-               "https://schema.management.azure.com/schemas/2018-05-01/subscriptionDeploymentTemplate.json#",
-               "https://schema.management.azure.com/schemas/2019-08-01/tenantDeploymentTemplate.json#",
-               "https://schema.management.azure.com/schemas/2019-08-01/managementGroupDeploymentTemplate.json#"
-        };
-
-        string[] validTemplateProperties = {
-                 "contentVersion",
-                 "apiProfile",
-                 "parameters",
-                 "variables",
-                 "functions",
-                 "resources",
-                 "outputs",
-        };
-
         /// <summary>
         /// Constructor for the command line parser. Sets up the command line API. 
         /// </summary>
         public CommandLineParser()
         {
             SetupCommandLineAPI();
```

Comment:
I think it would be a good idea to also refactor `SetupCommandLineAPI()` to do something like add all the new options to an array, that at the end of the function is used to add the options to both commands, so we don't have to remember to add new options into both every time

Reply:
I considered that too, but decided against it for some reason.  Since you also support this, I'll include this. 🙂